### PR TITLE
Fix/chat optimizations

### PR DESCRIPTION
**Pull request by kasenvr**
This PR closes the following issues: #469 #283 #201  #154 

TL;DR:

- Shrunk chat window.
- Space more efficiently used.
- Internal chat window now uses a more readable font (subject to change).
- Timestamp removes milliseconds but adds month and day (it's also now smaller and more readable).
- URLs can now be opened using an external browser.
- Add chat notification sound + mute function for it.
- The app moved folders.

In general, it shrinks the chat main window, optimizes the space inside to be closer together and use smaller fonts. The main chat window's messages also uses a clearer font than before. I might adjust that font again later, but for now this seems like a solid alternative. The timestamps are now smaller and out of the way, but also more readable. 

It also removes the millisecond stamping and in the new space created it adds the month and day so that chat timestamps are more useful.

The chat has also been moved to its appropriate folder "communityScripts" instead of "communityModules".

URLs for images, videos, and web pages in general now have a little icon next to them that launches it into your operating system's default browser. Copying and pasting of URLs now work without any extra code tagging along that you would have to remove in order to get a working URL normally.

Before:
image
After:
image

### CODE REVIEW

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-d5ec8e8_6c334005.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-d5ec8e8_6c334005.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-d5ec8e8.tar_6c334005.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-d5ec8e8.tar_6c334005.Z

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-d5ec8e8_6c334005.exe

File: scripts/communityScripts/chat/FloofChat.html [hunk 1]
```diff
@@ -171,13 +172,13 @@
                     case ext == "JPG":
                     case ext == "GIF":
                     case ext == "JPEG":
-                        elContent = elContent.replace(this, "<br/><img src='" + this + "'class=\"responsive z-depth-2\"><br/><a href=\"javascript:gotoClipboard('" + this + "');\" target='_blank'>" + this + "</a>");
+                        elContent = elContent.replace(this, "<br/><img src='" + this + "'class=\"responsive z-depth-2\"><br/><a href=\"javascript:gotoClipboard('" + this + "');\" target='_blank'>" + this + "</a><a onclick=\"gotoExternalURL('" + this + "');return false;\"href=\"" + this + "\">📲</a>");
```
Comment (ctrlaltdavid) [inline scripts/communityScripts/chat/FloofChat.html:175]: Lines too long: [2.1.4] File content must be kept within 128 columns.

File: scripts/communityScripts/chat/FloofChat.html [hunk 2]
```diff
@@ -284,6 +286,10 @@
     function gotoURL(url) {
         emitWebEvent({type: "CMD", cmd: "URL", url: url});
     }
+    
+    function gotoExternalURL(url) {
+        emitWebEvent({type: "CMD", cmd: "EXTERNALURL", url: url});
```
Comment (ctrlaltdavid) [inline scripts/communityScripts/chat/FloofChat.html:291]:
```suggestion
        emitWebEvent({ type: "CMD", cmd: "EXTERNALURL", url: url });
```

File: scripts/communityScripts/chat/FloofChat.js [hunk 3]
```diff
@@ -464,19 +470,22 @@ function messageReceived(channel, message) {
...
     var h = (d.getHours()).toString();
     var m = (d.getMinutes()).toString();
     var s = (d.getSeconds()).toString();
     var h2 = ("0" + h).slice(-2);
     var m2 = ("0" + m).slice(-2);
     var s2 = ("0" + s).slice(-2);
-    s2 += (d.getMilliseconds() / 1000).toFixed(2).slice(1);
-    return h2 + ":" + m2 + ":" + s2;
+    // s2 += (d.getMilliseconds() / 1000).toFixed(2).slice(1);
+    return month + "/" + day + " - " + h2 + ":" + m2 + ":" + s2;
```
Comment (ctrlaltdavid) [inline scripts/communityScripts/chat/FloofChat.js:482]: Hmmm. It might be nicer to use a user's local date format, e.g., so that there's no ambiguity over what 2/3 means. E.g., see Date.toLocaleDateString().

Comment (kasenvr) [reply scripts/communityScripts/chat/FloofChat.js:482]: Perhaps but the issue with that is that I have to add detection of the user's country and then figure out what that format will be. It might be easier to just follow ISO standards which is what I think I am doing now? It does it like so: MM/DD

Comment (ctrlaltdavid) [reply scripts/communityScripts/chat/FloofChat.js:482]: Date.toLocaleDateString() should work without needing any detection of user's country etc.

Comment (kasenvr) [reply scripts/communityScripts/chat/FloofChat.js:520]: You are right, I tried and it worked when I specify 'undefined', however it would not work at all no matter what options I specified. This caused a full date with even the day of the week to be outputted. It took up too much space then, and without a UI library like Vuetify, it'd be tough to find a way to fit in something like that without moving around things.

I figure that https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString was added pretty early, with all parameters being fully implemented by Chrome 24. I do not see why the QtScript engine would not have this implemented properly.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-dd33a47_c289434f.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-dd33a47.tar_c289434f.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-dd33a47.tar_c289434f.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-dd33a47_c289434f.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-dd33a47_c289434f.exe

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-8c15e58.tar_555dd8fe.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-8c15e58_555dd8fe.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-8c15e58.tar_555dd8fe.Z

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-8c15e58_555dd8fe.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-8c15e58_555dd8fe.exe

File: scripts/communityScripts/chat/FloofChat.js [hunk 4]
```diff
@@ -80,7 +85,7 @@ function init() {
     setupHistoryWindow(false);
 
     chatBar = new OverlayWindow({
-        source: Paths.defaultScripts + '/communityModules/chat/FloofChat.qml?' + Date.now(),
+        source: Paths.defaultScripts + '/communityScripts/chat/FloofChat.qml?' + Date.now(),
```
Comment (ctrlaltdavid) [inline scripts/communityScripts/chat/FloofChat.js:88]: It would be better to use Scripts.resolvePath() to get the QML relative to the script file rather than "hard-coded" Paths.defaultScripts value which evaluates to C:/Program Files/...etc. per install or build directory. (The latter makes it harder to test app changes during development.)

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-7d461a6.tar_c99a62b7.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-7d461a6_c99a62b7.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-7d461a6.tar_c99a62b7.Z

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-7d461a6_c99a62b7.exe

File: scripts/communityScripts/chat/FloofChat.js (hunk 4 above)
Comment (ctrlaltdavid) [reply scripts/communityScripts/chat/FloofChat.js:88]: Better code...
```
source: ROOT + "FloofChat.qml?" + Date.now(),
```

**Comment by Aitolda** [comment]
"Add chat notification sound + mute function for it."

Does not work in "grid" mode. No sound is heard even after unmuting.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-0ebdbb6_f74f0e8a.dmg

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-2c302d6.tar_fd2691fa.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-2c302d6.tar_fd2691fa.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-2c302d6_fd2691fa.sh

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-2c302d6_fd2691fa.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR485/Vircadia-Alpha-PR485-2c302d6_fd2691fa.exe
